object_database/web/content/CellHandler.js

```diff
@@ -140,113 +220,118 @@ class CellHandler {
 	    cell = cell.domNode;
 	}
 
-        if(message.discard !== undefined){
-            // In the case where we have received a 'discard' message,
-            // but the cell requested is not available in our
-            // cells collection, we simply display a warning:
-            if(cell == undefined){
-                console.warn(`Received discard message for non-existing cell id ${message.id}`);
-                return;
-            }
-	    // Instead of removing the node we replace with the a
-	    // `display:none` style node which effectively removes it
-	    // from the DOM
-	    if (cell.parentNode !== null) {
-		this.projector.replace(cell, () => {
-		    return h("div", {style: "display:none"}, []);
-		});
-	    }
-	} else if(message.id !== undefined){
-	    // A dictionary of ids within the object to replace.
-	    // Targets are real ids of other objects.
-	    let replacements = message.replacements;
-
-	    // TODO: this is a temporary branching, to be removed with a more logical setup. As
-	    // of writing if the message coming across is sending a "known" component then we use
-	    // the component itself as opposed to building a vdom element from the raw html
-	    let componentClass = this.components[message.component_name];
+
+	// A dictionary of ids within the object to replace.
+	// Targets are real ids of other objects.
+	let replacements = message.replacements;
+        var component = this.activeComponents[message.id];
+        if(component){
+            // In this case, this component has already been
+            // created and stored, so we just need to pass in
+            // updated props and re-render.
+            let nextProps = Object.assign({}, component.props, {
+                namedChildren: message.namedChildren,
+                children: message.children,
+                extraData: message.extraData
+            }, message.extraData);
+            let nextPropsToPass = component.componentWillReceiveProps(
+                component.props,
+                nextProps
+            );
+            component._updateReplacements(message.replacementKeys);
+            component._updateProps(nextPropsToPass);
+            //let cClass = this.components[message.componentName];
+            //var component = new cClass(nextProps, message.replacementKeys);
+            var velement = component.render();
+        } else {
+            // In this case, we are creating this cell for the first
```

Comment:
this is a `cellUpdated` type of message - why would there be a new cell to create here?

Reply:
There is no `cellCreated` message from the Python side at the moment. For now, the handler can easily tell whether or not we are updating an existing cell component or creating a new one based on the id's presence or absence from the `this.activeComponents` or even `this.cells` dictionaries.